combustion/gnns/Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM nvidia/cuda:11.3.1-cudnn8-runtime-ubuntu20.04
+
+# install required packages
+RUN apt-get update \
+ && apt-get install -y git \
+ python3.8-dev \
+ python3-pip \
+ build-essential \
+ libssl-dev \
+ libffi-dev
+
+
+RUN ln -s /usr/bin/python3.8 /usr/bin/python
+RUN pip install nox
+
+# create user ai4sim
+ARG uid=1000
+ARG gid=1000
+RUN groupadd -g ${gid} ai4sim \
+  && useradd -u ${uid} -g ${gid} -m -s /bin/bash ai4sim
+
+# change user
+USER ai4sim
+WORKDIR /home/ai4sim
+
+# clone ai4sim code
+RUN git clone https://github.com/AI4SIM/model-collection.git
```

Comment:
Clone the full repo in the image seem a bit overkill. Only the source of the use case could be embeded.
That could be done with a first temporary image used to retrive the source, and the final one will just copy the part it needs.

Reply:
The docker image built in our internal repo are good example of the method.

Reply:
The idea was to use a donwloader image (as done [here](https://bitbucketbdsfr.fsc.atos-services.net/projects/SBRA/repos/ai4sim-oci-images/browse/ubi8-mpi-hdf5/Dockerfile)) for all download/git clone/... stuffs.
Then only the part of relevant code (i.e. the use case related code) can be copied in the final image.
Please, note, it will require to remove the generic nox file at the /tools/nox path to copy/paste its content in the nox_file.py in each use-case, such that the divergence of each use-case will not break the nox sessions.